crates/matrix-sdk-base/src/client.rs

```diff
@@ -539,7 +539,17 @@ impl BaseClient {
     ) {
         for raw_event in events {
             if let Ok(event) = raw_event.deserialize() {
-                changes.add_room_account_data(room_id, event, raw_event.clone());
+                changes.add_room_account_data(room_id, event.clone(), raw_event.clone());
+
+                if let AnyRoomAccountDataEvent::MarkedUnread(e) = event {
+                    if let Some(room) = changes.room_infos.get_mut(room_id) {
+                        room.base_info.is_marked_unread = e.content.unread;
+                    } else if let Some(room) = self.store.get_room(room_id) {
+                        let mut info = room.clone_info();
+                        info.base_info.is_marked_unread = e.content.unread;
+                        changes.add_room(info);
+                    }
```

Comment:
What happens if the `room_id` doesn't exist in `changes.room_infos` nor `self.store`? Is it still possible?

Reply:
I think it is possible for e.g. rooms that are left but received account data changes for some reason but I think we can safely ignore those cases.

Reply:
Technically you can receive room account data for room in any state, so if you get invited again you won't receive the room account data in the next syncs.

Reply:
@stefanceriu Please add a comment saying that it's fine to ignore a `MarkedUnread` event when the room isn't know at this stage. It's not an error, it's just fine to ignore it.

Reply:
> Technically you can receive room account data for room in any state

Right so if you leave a room it will stay in the store and process account data correctly. If however the room isn't present in the current response or in the store it's basically unknown and the account data should be safe to ignore.